tensorboard/plugins/distribution/distributions_plugin.py

```diff
@@ -21,23 +36,29 @@
 from werkzeug import wrappers
 
 from tensorboard.backend import http_util
-from tensorboard.backend.event_processing import event_accumulator
 from tensorboard.plugins import base_plugin
-
-_PLUGIN_PREFIX_ROUTE = event_accumulator.COMPRESSED_HISTOGRAMS
+from tensorboard.plugins.distribution import compressor
+from tensorboard.plugins.histogram import histograms_plugin
 
 
 class DistributionsPlugin(base_plugin.TBPlugin):
-  """Distributions Plugin for TensorBoard."""
+  """Distributions Plugin for TensorBoard.
+
+  This supports both old-style summaries (created with TensorFlow ops
+  that output directly to the `histo` field of the proto) and new-style
+  summaries (as created by the `tensorboard.plugins.histogram.summary`
+  module).
+  """
 
-  plugin_name = _PLUGIN_PREFIX_ROUTE
+  plugin_name = 'distributions'
 
   def __init__(self, context):
     """Instantiates DistributionsPlugin via TensorBoard core.
 
     Args:
       context: A base_plugin.TBContext instance.
     """
+    self._histograms_plugin = histograms_plugin.HistogramsPlugin(context)
```

Comment:
I'm glad that you agree. I originally implemented this in two ways—once by copying the behavior of the histogram plugin, and once by encapsulating it like this. I was originally hesitant about including a hard dependency across plugins, but (a) Dandelion pointed out that this just reifies the existing data dependency in code and (b) it's _so_ much cleaner. It just feels _pleasant_ to read.